samples/testing-frameworks/appium/client-side/python/upload.py

```diff
@@ -38,14 +40,17 @@ def parse_args(self):
             sys.exit(1)
 
     def build_headers(self):
-        hdrs = {'Authorization': 'Basic %s' % base64.b64encode(self.api_key + ":"),
+        api_key =  base64.b64encode(bytes(self.api_key + ":",  encoding='utf8'))
+        hdrs = {'Authorization': 'Basic %s' % api_key,
                 'Accept': 'application/json'}
         return hdrs
 
     def upload_app(self):
         self.parse_args()
         files = {'file': (os.path.basename(self.myfile), open(self.myfile, 'rb'), 'application/octet-stream')}
-        r = requests.post(self.upload_url, files=files, headers=self.build_headers())
+        password = getpass.getpass("Enter host password for user '" + self.api_key +  "':")
+        r = requests.post(self.upload_url, files=files, headers=self.build_headers(),
+            auth=HTTPBasicAuth(self.api_key, password))
```

Comment:
Using api_key authorization causes 401 response error. Full authorization rosolves this problem.

Reply:
I see, it's not correct approach. It works, but there are a few issues in this solution:
- headers are not correct and they don't really matter, probably the header is overwritten with auth=...
- we ask for user input with getpass for no reason

It can be done this way:

```
    def build_headers(self):
        api_key = base64.standard_b64encode(bytes(self.api_key + ":", encoding='utf-8'))
        hdrs = {'Authorization': 'Basic %s' % api_key.decode('utf-8'), 'Accept': 'application/json'}
        return hdrs

    def upload_app(self):
        self.parse_args()
        files = {'file': (os.path.basename(self.myfile), open(self.myfile, 'rb'), 'application/octet-stream')}
        r = requests.post(self.upload_url, files=files, headers=self.build_headers())          
```

or simpler without headers:
```
    def upload_app(self):
        self.parse_args()
        files = {'file': (os.path.basename(self.myfile), open(self.myfile, 'rb'), 'application/octet-stream')}
        r = requests.post(self.upload_url, files=files, auth=HTTPBasicAuth(self.api_key, ''))
```

Reply:
Ok, thanks for help.